internal/stopwatch/timed.go

```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package stopwatch
+
+import (
+	"go.uber.org/atomic"
+	"time"
+)
+
+// Timed runs the given function and adds the time it took to the given atomic.Int64. Thread-safe
+func Timed(timer *atomic.Int64, f func()) {
+	start := time.Now()
+	f()
+	timer.Add(time.Since(start).Nanoseconds())
+}
+
+// TimedByStopwatch runs the given function and adds the time it took to the given stopwatch. NOT Thread-safe
+func TimedByStopwatch(watch Stopwatch, f func()) {
```

Comment:
This should 100% be a member of `Stopwatch` IMHO. You could have a design such as:

- `Stopwatch` has this `TimeFunc` method + embeds `stopwatchBase` interface that is your current `Stopwatch` methods
- A `stopwatch` struct embeds a `stopwatchBase` value + implements `TimeFunc`
- Voilà.